src/lib.rs

```diff
@@ -0,0 +1,339 @@
+mod network;
+
+#[allow(dead_code, unused_imports)]
+
+pub type Meters = f32;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum End {
+    Front,
+    Back,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Direction {
+    Forward,
+    Backward,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Directions {
+    Forward,
+    Backward,
+    BothWays,
+    Alternating,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Side {
+    Left,
+    Right,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SideDirection {
+    Leftward,
+    Rightward,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum LaneSide {
+    Inside,
+    Outside,
+}
+
+pub enum LaneDirection {
+    Inward,
+    Outward,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum DrivingSide {
+    LHT,
+    RHT,
+}
+
+use Direction::*;
+use DrivingSide::*;
+use End::*;
+use LaneDirection::*;
+use LaneDirection::*;
+use LaneSide::*;
+use Side::*;
+use SideDirection::*;
+
+impl DrivingSide {
+    fn get_direction(&self, side: Side) -> Direction {
+        match (self, side) {
+            (Self::LHT, Side::Left) | (Self::RHT, Side::Right) => Direction::Forward,
+            (Self::LHT, Side::Right) | (Self::RHT, Side::Left) => Direction::Backward,
+        }
+    }
+    fn get_side(&self, dir: Direction) -> Side {
+        match (self, dir) {
+            (Self::LHT, Direction::Forward) | (Self::RHT, Direction::Backward) => Side::Left,
+            (Self::LHT, Direction::Backward) | (Self::RHT, Direction::Forward) => Side::Right,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Carriage {
+    Truck,
+    Bus,
+    Car,
+    Bike,
+    Foot,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Designation {
+    /// Think "carriageway" carriages, anything from trucks to mopeds to drawn carts.
+    Travel(Carriage),
+    /// Loading zones for trucks too, with short stay?
+    Parking(Carriage),
+    /// Verge and stuff, those outdoor eating areas
+    Amenity,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum RoadEdge {
+    /// Not an edge at all, but a continuation into more road surface.
+    /// Expect Join to a Buffer for railings/bollards.
+    Join,
+    /// The road just ends and transitions into another groundcover.
+    Sudden,
+    /// A short rise up from the road surface. Constructed, usually out of concrete in a certain shape.
+    Kerb,
+    /// A cliff or the edge of a bridge.
+    Drop,
+    /// Walls etc. that interrupt the road surface.
+    Barrier,
+}
+
+/// A single lane on the carriageway, with designation, width, etc.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Lane {
+    dir: Directions,
+    designation: Designation,
+    width: Meters,
+    can_enter_from_inside: bool,
```

Comment:
What's inside/outside in the context of lanes possibly with alternating/multiple directions?

Reply:
relative to the roadway, so `can_enter_from_road_inside: bool` might be better. I think I'll change `LaneSide` to `RoadSide`...